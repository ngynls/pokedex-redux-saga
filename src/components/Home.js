import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as actions from '../store/actions/actions';
import PokedexElement from './PokedexElement';

class Home extends Component{

    componentDidMount(){
       this.props.fetchPokemons();
    }

    render(){
        const pokemonList=this.props.pokemons.map( (pokemon, i) => {
            return (
                <div className="col-xl-4 col-lg-6 col-md-4" key={i+1}>
                    <PokedexElement key={i+1} id={i+1} pokemon={pokemon} />
                </div>
            );
        });
        return(
            <div className="container">
                <div className="row">
                    {
                        pokemonList
                    }
                </div>
            </div>
        );
    }
}

function mapStateToProps(state){
    return {
        pokemons: state.pokemonReducer.pokemons,
        loading: state.pokemonReducer.loading,
        error: state.pokemonReducer.error
    }
}

function mapDispatchToProps(dispatch){
    return {
        fetchPokemons: ()=> dispatch(actions.fetch_pokemon_requested()),
        searchPokemonByName: (name)=>dispatch(actions.search_by_name(name))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);