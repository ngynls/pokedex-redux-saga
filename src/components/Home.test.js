import React from "react";
import renderer from "react-test-renderer";
import Home from "./Home";
import rootReducer from "../store/reducers";
import { Provider } from "react-redux";
import { createStore } from "redux";

test("matches snapshot of Home component", () => {
    const store = createStore(rootReducer);
    const tree = renderer.create(<Provider store={ store }><Home /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});