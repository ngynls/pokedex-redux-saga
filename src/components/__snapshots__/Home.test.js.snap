// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot of Home component 1`] = `
<div
  className="container"
>
  <div
    className="search-filter-bar mt-4"
  >
    <input
      aria-label="Search"
      className="form-control"
      onChange={[Function]}
      placeholder="Search for a Pokemon"
      type="search"
    />
  </div>
  <div
    className="row"
  />
</div>
`;
